grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -86,6 +87,34 @@ class PicklistService {
         productAvailabilityService.refreshProductsAvailability(orderItem?.order?.origin?.id, [orderItem?.product?.id], binLocations?.unique(), false)
     }
 
+    void clearPicklist(String id) {
+        Picklist picklist = Picklist.get(id)
+
+        if (!picklist) {
+            throw new ObjectNotFoundException(id, Picklist.class.toString())
+        }
+        // Store bin locations' id for product availability refresh
+        List<String> binLocations = []
+
+        Set<PicklistItem> picklistItems = picklist.picklistItems
+        List<PicklistItem> itemsToRemove = []
+        picklistItems.each { PicklistItem picklistItem ->
+            picklistItem.disableRefresh = true
+            itemsToRemove.add(picklistItem)
+            binLocations.add(picklistItem.binLocation?.id)
+        }
+        itemsToRemove.each {
+            picklist.removeFromPicklistItems(it)
+        }
+
+
+        productAvailabilityService.refreshProductsAvailability(
+                picklist?.requisition?.origin?.id,
+                picklist?.requisition?.requisitionItems?.product?.id,
+                binLocations.unique(),
```

Comment:
Make binLocations a Set<String> and we probably don't need to deal with the unique issue.

Reply:
It can't be a Set, because the product availability method requires a List, and it breaks, if the binLocations become the Set. I think refactoring it now (the PA method), or creating another overloaded method with Set instead of List does not make sense, unless you feel strong about it.